File: medico/Services/Medico.Api/ClientApp/src/app/patientChart/models/medicationHistory.ts
export class MedicationHistory {
    id: string;
    createDate: any;
    medication: string;
    patientId: string;
    medicationNameId: string;
    units: string;
    dose: string;
    route: string;
    prn: boolean;
    medicationStatus: string;
    notes: string;
    dosageForm: string;
    sig: string;

    constructor(patientId: string = "") {
        this.createDate = new Date();
        this.prn = false;
        this.patientId = patientId;
    }
}

File: medico/Domain/Medico.Domain/Models/MedicationHistory.cs
using System;

namespace Medico.Domain.Models
{
    public class MedicationHistory : Entity
    {
        public DateTime? CreateDate { get; set; }
        public string Medication { get; set; }
        public Guid? MedicationNameId { get; set; }
        public MedicationName MedicationName { get; set; }
        public Patient Patient { get; set; }
        public Guid PatientId { get; set; }
        public string Dose { get; set; }
        public string Units { get; set; }
        public string DosageForm { get; set; }
        public string Route { get; set; }
        
        public string Sig { get; set; }

        public bool? Prn { get; set; }
        public string MedicationStatus { get; set; }
        public string Notes { get; set; }
    }
}